algorithms/data_structures/lcp_array.py

```diff
@@ -0,0 +1,86 @@
+import math
+
+"""
+    Suffix Array
+    ------------------
+    In computer science, a suffix array is a sorted array of all suffixes
+    of a string. It is a data structure used, among others, in full text
+    indices, data compression algorithms and  within the field
+    of bioinformatics.
+
+    for more info : http://algs4.cs.princeton.edu/63suffix/
+    Complexity :
+        worst case : O(n log(n))
+"""
+
+
+def suffix_array(t_string):
```

Comment:
Can you add doc strings? Take a look at few of the other modules to get an idea of what should be expected.